python-client/giskard/ml_worker/testing/registry/giskard_test.py

```diff
@@ -109,14 +109,18 @@ def __init__(self, test_fn: Function) -> None:
             meta = tests_registry.get_test(test_uuid)
         super(GiskardTest, self).__init__(test_fn, meta)
 
-    def __call__(self, *args, **kwargs) -> 'GiskardTestMethod':
-        self.is_initialized = True
-        self.params = kwargs
+    def __call__(self, *args, **kwargs) -> Result:
+        return self.test_fn(*args, **kwargs)
+
+    def set_params(self, *args, **kwargs) -> 'GiskardTestMethod':
+        new_instance = GiskardTestMethod(self.data)
+        new_instance.is_initialized = True
+        new_instance.params = kwargs
 
         for idx, arg in enumerate(args):
-            self.params[next(iter([arg.name for arg in self.meta.args.values() if arg.argOrder == idx]))] = arg
+            new_instance.params[next(iter([arg.name for arg in self.meta.args.values() if arg.argOrder == idx]))] = arg
```

Comment:
It’s a bit strange to manipulate the object attribute `params` from outside.

Also, this is not clear:
```
next(iter([arg.name for arg in self.meta.args.values() if arg.argOrder == idx]))
```
I think `iter` is not necessary, could just get the first item if that’s what we need. By why not getting the list of arg names (e.g. by inspection) and then just doing `arg_name[n]`?